app/Database/src/Database.php

```diff
@@ -8,36 +8,44 @@
 
 namespace FlightSim\Database;
 
-class Database extends \PDO {
-
-  public function __construct() {
-  }
-
-  /**
-   * Creates PDO object to be used to interact with the database through out the
-   * application.
-   */
-  public function connect() {
-    $info = $this->getConnectionInfo();
-    try {
-      parent::__construct('mysql://' . $info['name'] . ':' . $info['password'] . '@' . $info['uri'] . '/' . $info['database']);
-    } catch (\PDOException $e) {
-      sprintf('Database connection failed: %s', $e->getMessage());
+class Database {
+
+    public $xml;
+
+    public function __construct()
+    {
+        $xml = file_get_contents('app/Database/data/database.xml');
+        $this->xml = new \SimpleXMLElement($xml);
+    }
+
+    // @todo Load a certain object of a certain type.
+    public function load($type, $identifier)
+    {
+
+        $result = $this->xml->xpath(
+          // @todo Evaluate if it's reliable that $type(s)/$type will always work.
```

Comment:
We could validate types at a higher level. @ramorgan and I have talked about making certain variables enums. Which are more or less a less of predefined terms. So that when entering the terms upstream we can validate that they are correct.